DXMainClient/DXGUI/Multiplayer/PlayerExtraOptionsPanel.cs

```diff
@@ -0,0 +1,315 @@
+            btnHelp.ClientRectangle = new Rectangle(chkBoxUseTeamStartMappings.Right + 4, chkBoxUseTeamStartMappings.Y - 1, 0, 0);
+            AddChild(btnHelp);
+
+            var lblPreset = new XNALabel(WindowManager);
+            lblPreset.Name = nameof(lblPreset);
+            lblPreset.Text = "Presets:";
+            lblPreset.ClientRectangle = new Rectangle(chkBoxUseTeamStartMappings.X, chkBoxUseTeamStartMappings.Bottom + 8, 0, 0);
+            AddChild(lblPreset);
+
+            ddTeamStartMappingPreset = new XNAClientDropDown(WindowManager);
+            ddTeamStartMappingPreset.Name = nameof(ddTeamStartMappingPreset);
+            ddTeamStartMappingPreset.ClientRectangle = new Rectangle(lblPreset.X + 50, lblPreset.Y - 2, 160, 0);
+            ddTeamStartMappingPreset.SelectedIndexChanged += DdTeamMappingPreset_SelectedIndexChanged;
+            ddTeamStartMappingPreset.AllowDropDown = true;
+            AddChild(ddTeamStartMappingPreset);
+
+            teamStartMappingsPanel = new TeamStartMappingsPanel(WindowManager);
+            teamStartMappingsPanel.ClientRectangle = new Rectangle(200, ddTeamStartMappingPreset.Bottom + 8, Width, Height - ddTeamStartMappingPreset.Bottom + 4);
+            AddChild(teamStartMappingsPanel);
+
+            AddLocationAssignments();
+
+            base.Initialize();
+
+            RefreshTeamStartMappingsPanel();
+        }
+
+        private void BtnHelp_LeftClick(object sender, EventArgs args)
+        {
+            XNAMessageBox.Show(WindowManager, "Team/Start Mappings",
+                "Team/start mappings allow the host to assign starting locations to teams, not players.\n" +
+                "When used with random teams/starts, players will be auto allied based on where they are spawned.\n" +
+                $"You can \"block\" a location from being randomly assigned using \"{TeamStartMapping.BLOCK}\" as the team.\n" +
+                "All locations must be assigned or blocked.\n"
+            );
+        }
+
+        public void UpdateForMap(Map map)
+        {
+            if (_map == map)
+                return;
+
+            _map = map;
+
+            RefreshTeamStartMappingPanels();
+        }
+
+        public List<TeamStartMapping> GetTeamStartMappings()
+        {
+            return chkBoxUseTeamStartMappings.Checked ? teamStartMappingsPanel.GetTeamStartMappings() : new List<TeamStartMapping>();
+        }
+
+        public void EnableControls(bool enable)
+        {
+            chkBoxForceRandomSides.InputEnabled = enable;
+            chkBoxForceRandomColors.InputEnabled = enable;
+            chkBoxForceRandomStarts.InputEnabled = enable;
+            chkBoxForceRandomTeams.InputEnabled = enable;
+            chkBoxUseTeamStartMappings.InputEnabled = enable;
+
+            teamStartMappingsPanel.EnableControls(enable && chkBoxUseTeamStartMappings.Checked);
+        }
+
+        public PlayerExtraOptions GetPlayerExtraOptions() =>
+            new PlayerExtraOptions()
```

Comment:
IMHO it's better to keep `=>` at the second line because
a) in case the function is gonna be expanded the line with signature won't need to be changed from git standpoint
b) it's easier to see such stuff at the beginning of a line
c) it's logically related to the body of the function more

Reply:
No input on this one?